exporters/prometheus/src/main/java/io/opentelemetry/exporters/prometheus/MetricAdapter.java

```diff
@@ -0,0 +1,192 @@
+        labelNames.addAll(constLabelNames);
+        labelValues =
+            new ArrayList<>(descriptor.getConstantLabels().size() + point.getLabels().size());
+        labelValues.addAll(constLabelValues);
+
+        for (Map.Entry<String, String> entry : point.getLabels().entrySet()) {
+          // TODO: Use a cache(map) of converted label names to avoid sanitization multiple times
+          // for the same label key.
+          labelNames.add(toLabelName(entry.getKey()));
+          labelValues.add(entry.getValue() == null ? "" : entry.getValue());
+        }
+      }
+
+      switch (descriptor.getType()) {
+        case MONOTONIC_DOUBLE:
+        case NON_MONOTONIC_DOUBLE:
+          DoublePoint doublePoint = (DoublePoint) point;
+          samples.add(new Sample(name, labelNames, labelValues, doublePoint.getValue()));
+          break;
+        case MONOTONIC_LONG:
+        case NON_MONOTONIC_LONG:
+          LongPoint longPoint = (LongPoint) point;
+          samples.add(new Sample(name, labelNames, labelValues, longPoint.getValue()));
+          break;
+        case SUMMARY:
+          SummaryPoint summaryPoint = (SummaryPoint) point;
+          samples.add(
+              new Sample(
+                  name + SAMPLE_SUFFIX_COUNT, labelNames, labelValues, summaryPoint.getCount()));
+          samples.add(
+              new Sample(name + SAMPLE_SUFFIX_SUM, labelNames, labelValues, summaryPoint.getSum()));
+          List<ValueAtPercentile> valueAtPercentiles = summaryPoint.getPercentileValues();
+          List<String> labelNamesWithQuantile = new ArrayList<>(labelNames.size());
+          labelNamesWithQuantile.addAll(labelNames);
+          labelNamesWithQuantile.add(LABEL_NAME_QUANTILE);
+          for (ValueAtPercentile valueAtPercentile : valueAtPercentiles) {
+            List<String> labelValuesWithQuantile = new ArrayList<>(labelValues.size());
+            labelValuesWithQuantile.addAll(labelValues);
+            labelValuesWithQuantile.add(doubleToGoString(valueAtPercentile.getPercentile()));
+            samples.add(
+                new Sample(
+                    name,
+                    labelNamesWithQuantile,
+                    labelValuesWithQuantile,
+                    valueAtPercentile.getValue()));
+          }
+          break;
+      }
+    }
+    return samples;
+  }
+
+  // Converts a label keys to a label names. Sanitizes the label keys.
+  static String toLabelName(String labelKey) {
+    return Collector.sanitizeMetricName(labelKey);
+  }
+
+  private static int estimateNumSamples(int numPoints, Descriptor.Type type) {
+    switch (type) {
+      case NON_MONOTONIC_LONG:
+      case NON_MONOTONIC_DOUBLE:
+      case MONOTONIC_LONG:
+      case MONOTONIC_DOUBLE:
+        return numPoints;
+      case SUMMARY:
+        // count + sum + estimated 2 percentiles (default MinMaxSumCount aggregator).
+        return numPoints * 4;
+    }
+    return numPoints;
```

Comment:
not sure if it might read better to have `return numPoints;` as the `default` block in the switch.